### ISSUE TITLE
ECS driver: More user friendly API design 

### ISSUE THREAD

**Issue by denverdino** [denverdino/aliyungo#4]
1. Current API design is just mapping of the REST API. It will be better to provide the resource base API. E.g. instance resource, disk resource we can associate the disk resource to instance

2. The WaitForXXX method can be simplified.

**Comment by AVVS** [comment]
I'd also ask to use interfaces instead of structs, because its impossible to extend them in the underlaying libraries (ie in terraform provider where its used). Often it lacks some of the supported attributes and forking / copying source to extend is not a maintainable option =(

Either way thanks for the library ;)

**Comment by denverdino** [comment]
@AVVS Thanks for your comments. We will consider that.

And the official SDK for Aliyun services is published. They are generated from Alibaba Cloud API. Please visit https://github.com/aliyun/alibaba-cloud-sdk-go  and any comments are welcome.

**Comment by AVVS** [comment]
afaik https://github.com/alibaba/terraform-provider doesn't use the official SDK, so if that changes, it certainly would solve the problem ;) but so far its kind of impossible not to use this library

**Comment by denverdino** [comment]
Right, that is the timing issue. And you can open issue for the terraform provider.  I will let my colleague to follow it up. Thanks again
